### PR TITLE
Ensure ESLint config is being applied as expected

### PR DESCRIPTION
**Pull request by osg74**
### Overview
The `lint` and `lint-fix` Makefile targets don't currently seem to detect and enforce the `.eslintrc.js` config as expected.  To fix this with the current `xargs eslint` implementation we need to explicitly pass the eslint config as well.

A further improvement would be to call `eslint` directly instead -- we can then rely on the corrected `.eslintignore` file rather than piped `grep` calls for eliminating irrelevant files, e.g.:

```
${ROOTDIR}/node_modules/.bin/eslint -c .eslintrc.js --ext .ts .
```

### Verification
- The original bug was discovered when running `eslint` with the `--debug` option and the subsequent fixes were also verified this way along with explicit result output using the `--format tap` option.
- Final build results for these changes are also [green](https://app.circleci.com/pipelines/github/osg74/packs-sdk/18/workflows/25b8137e-a77b-4cc7-9f22-9acd94074aef/jobs/21/steps).

### CODE REVIEW

**Comment by osg74** [comment]
I know the Contribution guide mentions that SDK PRs are not accepted 🙈 but thought this might be a fix worth sharing anyway.

File: .eslintignore [hunk 1]
```diff
@@ -1,5 +1,12 @@
...
+.idea
+.vscode
+.yarn*
+dist
+node_modules
+*.d.ts
+*.js
+
+# This has a separate self-contained .eslintrc.js configuration
+documentation/samples
```
Comment (ekoleda-codaio) [inline .eslintignore:12]: There is a separate config for these files, but if you add this line then those files aren't linted at all. In the main branch, running `make lint` catches errors in those files, using the local config. This should be removed.

File: .gitignore [hunk 2]
```diff
@@ -1,15 +1,12 @@
-.coda
-.coda.json
+*yarn*
```
Comment (ekoleda-codaio) [inline .gitignore:1]: Are these changes relevant to PR? It would seem tangential to update this file. Additionally, some of these wildcards seem too broad.

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ This changelog keeps track of all changes to the Packs SDK. We follow convention
 ### Changed
 
 - Updated `PrecannedDateRange` to match the date range picker in the Coda UI: added `Last90Days`, `Last180Days`, `Last365Days`, `Next90Days`, `Next180Days`, `Next365Days`, `Last7AndNext7Days`, `Last30AndNext30Days`, deprecated `Last3Months`, `Last6Months`, `Next3Months`, and `Next6Months`, and removed `ThisWeekStart`, `ThisMonthStart`, and `ThisYearStart`, which never actually worked.
+- Improved `eslint` calls in `Makefile` to ensure the associated ESLint config files are applied as expected.
```
Comment (ekoleda-codaio) [inline CHANGELOG.md:10]: We tend to only include changes in the changelog if consumers of the SDK will notice them. A change to the eslint config only affects maintainers, so I'd omit it.

File: Makefile [hunk 4]
```diff
@@ -81,7 +81,7 @@ bootstrap:
 
 .PHONY: lint
 lint:
-	find . -name "*.ts" | grep -v /dist/ | grep -v /node_modules/ | grep -v "\.d\.ts" | xargs ${ROOTDIR}/node_modules/.bin/eslint
+	${ROOTDIR}/node_modules/.bin/eslint -c .eslintrc.js -f tap --ext .ts .
```
Comment (ekoleda-codaio) [inline Makefile:84]: While including `-f tap` may be useful for testing, it changes the output of this make rule and should probably be removed. Here and below.

File: .eslintignore [hunk 5]
```diff
@@ -1,5 +1,12 @@
-.circleci/*
-.githooks/*
-.idea/*
-.vscode/*
-node_modules/*
+.circleci
+.githooks
+.idea
+.vscode
+.yarn*
```
Comment (ekoleda-codaio) [inline .eslintignore:5]: What is this meant to match against?

File: .eslintignore [hunk 6]
```diff
@@ -1,5 +1,12 @@
...
+dist
+node_modules
+*.d.ts
```
Comment (ekoleda-codaio) [inline .eslintignore:8]: Why wouldn't we want to lint these?

File: .eslintignore [hunk 7]
```diff
@@ -1,5 +1,12 @@
...
+*.js
```
Comment (ekoleda-codaio) [inline .eslintignore:9]: While this likely to be true going forward, I think it would be safer to just list the files to ignore, since there are a small number.

**Review by ekoleda-codaio — CHANGES_REQUESTED** [review]
Thanks for wanting to contribute. I have noticed that VSCode gets a little angry about certain config files not matching these rules, so there certainly is a problem to be solved.

File: CHANGELOG.md (hunk 3 above)
Comment (osg74) [reply CHANGELOG.md:10]: OK, good to know -- I'll remove this.

File: Makefile (hunk 4 above)
Comment (osg74) [reply Makefile:84]: This was included more for illustration, but I'll remove the explicit output for the final version.

File: .eslintignore (hunk 6 above)
Comment (osg74) [reply .eslintignore:8]: Agreed, though the original lint command in the Makefile excluded *.d.ts files in the various grep filters, hence the addition here.  I'll remove line 8 in this case.

File: .eslintignore (hunk 5 above)
Comment (osg74) [reply .eslintignore:5]: Prevents parsing of any yarn files and esp. the .yarn directory though passing an `--ext .ts` option to the `eslint` command is arguably more efficient.

File: .eslintignore (hunk 7 above)
Comment (osg74) [reply .eslintignore:9]: The linter gave some warnings about parsing .js files when it wasn't expecting them but, as above, adding `--ext .ts` seems to operate better than using an ignore statement in this case.

File: .eslintignore (hunk 1 above)
Comment (osg74) [reply .eslintignore]: Indeed -- I'll remove this as it was added in error along with a `root` directive in the lower level eslint config.

File: .gitignore (hunk 2 above)
Comment (osg74) [reply .gitignore:1]: I'll revert this.

**Comment by osg74** [comment]
I'll cancel this PR and rather submit a cleaned up version based on the recommendations discussed above.
